### PR TITLE
Fixes the majority of golint issues

### PR DESCRIPTION
**Pull request by jterry75**
Partial: #12

### CODE REVIEW

File: service/gcs/errors/errors.go [hunk 1]
```diff
@@ -29,9 +35,9 @@ func (e *containerExistsError) Error() string {
 	return fmt.Sprintf("a container with the ID \"%s\" already exists", e.ID)
 }
 
-// NewContainerExistsError returns a *containerExistsError referring to the
+// NewContainerExistsError returns an error referring to the
```
Comment (beweedon) [inline service/gcs/errors/errors.go]: nit: this and some of the other doc comments below can be reduced to a single 80-character line now.

File: service/gcs/oslayer/realos/realos.go [hunk 2]
```diff
@@ -21,9 +21,9 @@ type realProcessExitState struct {
 	state *os.ProcessState
 }
 
-// NewProcessExitState returns a *realProcessExitState wrapping the given
+// NewProcessExitState returns a oslayer.ProcessExitState wrapping the given
```
Comment (beweedon) [inline service/gcs/oslayer/realos/realos.go]: nit: "an oslayer.ProcessExitState"

File: service/gcs/prot/protocol.go [hunk 3]
```diff
@@ -38,9 +38,14 @@ const (
 type MessageType uint32
 
 const (
-	MtNone         = 0
-	MtRequest      = 0x10000000
-	MtResponse     = 0x20000000
+	// MtNone is the default MessageHeader.Type.
```
Comment (beweedon) [inline service/gcs/prot/protocol.go]: nit: Replace "MessageHeader.Type" with "MessageType", same below?

File: service/gcs/runtime/runc/utils_test.go [hunk 4]
```diff
@@ -17,11 +18,13 @@ var _ = Describe("Utils", func() {
 	)
 
 	BeforeEach(func() {
-		rtime, err = NewRuntime()
+		var rt runtime.Runtime
+		rt, err = NewRuntime()
```
Comment (beweedon) [inline service/gcs/runtime/runc/utils_test.go]: You can probably replace the `=` with a `:=` here, and delete the `var rt runtime.Runtime` line.

File: service/gcs/errors/errors.go [hunk 5]
```diff
@@ -11,13 +11,19 @@ import (
...
-	HrInvalidArg   = Hresult(-2147024809) // 0x80070057
-	HrPointer      = Hresult(-2147467261) // 0x80004003
-	HrFail         = Hresult(-2147467259) // 0x80004005
+	// HrUnexpected is the HRESULT for an unexpected exit.
+	HrUnexpected = Hresult(-2147418113) // 0x8000FFFF
+	// HrNotImpl is the HRESULT for a not implemented function.
+	HrNotImpl = Hresult(-2147467263) // 0x80004001
+	// HrInvalidArg is the HRESULT for an invalid argument to a function.
+	HrInvalidArg = Hresult(-2147024809) // 0x80070057
+	// HrPointer is the HRESULT for a nil pointer that is required to a function.
```
[the remark this answers is not in the document]
Comment (jterry75) [reply service/gcs/errors/errors.go]: That would be a `panic()`. It really is for an expectation right?

File: service/gcs/errors/errors.go (hunk 1 above)
Comment (jterry75) [reply service/gcs/errors/errors.go]: Fixed.

File: service/gcs/oslayer/realos/realos.go (hunk 2 above)
Comment (jterry75) [reply service/gcs/oslayer/realos/realos.go]: Fixed.

File: service/gcs/prot/protocol.go (hunk 3 above)
Comment (jterry75) [reply service/gcs/prot/protocol.go]: Done.

File: service/gcs/runtime/runc/utils_test.go (hunk 4 above)
Comment (jterry75) [reply service/gcs/runtime/runc/utils_test.go]: Fixed.

File: service/gcs/errors/errors.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (beweedon) [reply service/gcs/errors/errors.go]: Oh, oops, you're right! My bad! This seems fine :)

[the remark this answers is not in the document]
Comment (beweedon) [reply service/gcs/errors/errors.go]: Maybe replace "required" with "passed", though? I wasn't sure I understood the comment before.

**Comment by beweedon** [comment]
Looks good to me! Feel free to address my comment about the wording of HrPointer if you want, but also up to you!

File: service/gcs/errors/errors.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (jterry75) [reply service/gcs/errors/errors.go]: Ah ok. Done

**Pull request merged**